openedx_events/tooling.py

```diff
@@ -0,0 +1,189 @@
+"""
+Tooling necessary to use Open edX events.
+"""
+import socket
+import warnings
+from datetime import datetime
+from uuid import uuid1
+
+from django.conf import settings
+from django.dispatch import Signal
+
+import openedx_events
+from openedx_events.exceptions import InstantiationError, SenderValidationError
+
+
+class OpenEdxPublicSignal(Signal):
+    """
+    Custom class used to create Open edX events.
+    """
+
+    def __init__(self, event_type, data, minor_version=0):
+        """
+        Init method for OpenEdxPublicSignal definition class.
+
+        Arguments:
+            event_type (str): name of the event.
+            data (dict): attributes passed to the event.
+            minor_version (int): version of the event type.
+        """
+        if not event_type:
+            raise InstantiationError(
+                message="Missing required argument 'event_type'"
+            )
+        if not data:
+            raise InstantiationError(
+                event_type=event_type, message="Missing required argument 'data'"
+            )
+        self.init_data = data
+        self.event_type = event_type
+        self.minor_version = minor_version
+        super().__init__()
+
+    def __repr__(self):
+        """
+        Represent OpenEdxPublicSignal as a string.
+        """
+        return "<OpenEdxPublicSignal: {event_type}>".format(event_type=self.event_type)
+
+    def generate_signal_metadata(self):
```

Comment:
Hey @felipemontoya! what do you think about changing `generate_signal_metadata` for [this](https://github.com/eduNEXT/openedx-events/pull/17/files#diff-327e7d7bba7d257820e4e319a740dbbd12556734420a7c8df0e42583ebea6792L49) version? It uses EventsMetadata as an attr class

Reply:
I like the idea of keeping the in memory data in a more natural python form and I think that the attr class for events metadata makes it well defined and robust. I'm happy to see that it was adopted already.